units/commit_queue.go

```diff
@@ -66,6 +66,32 @@ func NewCommitQueueJob(env evergreen.Environment, queueID string, id string) amb
 	return job
 }
 
+func (j *commitQueueJob) TryUnstick(cq *commitqueue.CommitQueue) {
+	//unstuck the queue if the patch is done.
+	nextItem := cq.Next()
+	if nextItem != nil {
+		patch, err := patch.FindOne(patch.ById(patch.NewId(nextItem.Issue)).WithFields(patch.FinishTimeKey))
+		if err != nil {
+			j.AddError(errors.Wrapf(err, "error determining if patch is done for %s", j.QueueID))
+		}
+
+		//patchisdone
+		if !patch.FinishTime.IsZero() {
+			j.dequeue(cq, nextItem)
```

Comment:
I think we should also log the event for the merge completing, like we do [here](https://github.com/evergreen-ci/evergreen/blob/8c915b9282ad7b4c36a0be3182398c20f6d713d2/model/commitqueue/dequeue_sender.go#L58).

Reply:
done